Diplom_3

Steamin01 opened the pull request.

Steamin01 committed:
UiTests

okryukova commented on src/main/java/pages/ConstructorPage.java:
```diff
@@ -0,0 +1,33 @@
+
+@Data
+@AllArgsConstructor
+public class ConstructorPage {
```
⛔️Нужно исправить.  Это часть MainPage. Локаторы дублируются

okryukova commented on src/test/java/ConstructorTest.java:
```diff
@@ -0,0 +1,46 @@
+    @Step("Открыть раздел «Соусы» с главной страницы")
+    private void goToSaucesSection() {
+        MainPage main = new MainPage(driver);
+        main.openSaucesSection();
```
⛔️Нужно исправить.  В каждом тесте должен быть ассерт. Сейчас тест ничего не проверяет.

okryukova commented on src/test/java/LoginUserTest.java:
```diff
@@ -0,0 +1,102 @@
+public class LoginUserTest extends BaseTest {
+    private WebDriverWait wait;
+    private final String email = "MiuMiu@mail.ru";
+    private final String pass = "MiuMiuMiu";
```
⛔️Нужно исправить. Согласно заданию, тестового пользователя нужно создать с помощью АПИ запроса. Это должен быть рандомный пользователь, уникальный для каждого теста. После выполнения теста его нужно удалить.  Поправь, пожалуйста, этот момент во всем коде

okryukova commented on src/test/java/LoginUserTest.java:
```diff
@@ -0,0 +1,102 @@
+    }
+
+    @Test
+    @DisplayName("Вход из формы восстановления пароля")
```
⚠️Можно улучшить. В allure есть еще одна аннотация для описания теста Description. Добавь ее, пожалуйста, во всех тестах проекта

okryukova commented on src/test/java/LoginUserTest.java:
```diff
@@ -0,0 +1,102 @@
+
+    @Step("Проверить, что пользователь успешно авторизовался")
+    private void assertLoggedIn() {
+        By checkoutButtonLocator = By.xpath("//button[text()='Оформить заказ']");
```
⛔️Нужно исправить. Не следует описывать локаторы внутри тестовых методов. Используй инкапсуляцию. В классах page objects опиши все веб-элементы, которые используются в кейсах, с помощью локаторов и укажи модификатор доступа private. Реализуй шаги для действий над веб-элементами. Тесты должны состоять из шагов (для получения актуального результата) и проверки (сравнение ожидаемого и актуального результата). Это позволит переиспользовать веб-элементы в разных шагах(методах), а шаги в разных тестах.

Steamin01 committed:
Fix#1
- ConstructorPage был удалён
- Добавлен ForgotPasswordPage
- В MainPage добавлены дополнительные Step
- Во всех тестах был добавлен Assert
- Добавлены User.java и UserApi.java для создания тестового пользователя через Api
- Для всех тестов было добавлено описание Description
- Локаторы внутри тестовых методов удалены
